model/sharing/shared.go

```diff
@@ -263,14 +263,15 @@ func updateRemovedForFiles(inst *instance.Instance, sharingID, dirID string, rul
 // UpdateShared updates the io.cozy.shared database when a document is
 // created/update/removed
 func UpdateShared(inst *instance.Instance, msg TrackMessage, evt TrackEvent) error {
-	mu := lock.ReadWrite(inst, "shared")
+	evt.Doc.Type = msg.DocType
+	sid := evt.Doc.Type + "/" + evt.Doc.ID()
+
+	mu := lock.ReadWrite(inst, "shared/"+sid)
```

Comment:
If we update the lock name here, we should also look at the other calls to the same lock, like https://github.com/cozy/cozy-stack/blob/master/model/sharing/upload.go#L342

Reply:
Right, I updated the lock in 2 more places. I just left this one https://github.com/cozy/cozy-stack/blob/master/model/sharing/setup.go#L206